src/fides/api/service/connectors/saas_connector.py

```diff
@@ -275,29 +282,75 @@ def retrieve_data(
-                        read_request,
+                # Iterates through initial list of prepared requests and through subsequent
+                # requests generated by pagination. The results are added to the output
+                # list of rows after each request.
+                for next_request, param_value_map in prepared_requests:
+                    while next_request:
+                        processed_rows, next_request = self.execute_prepared_request(  # type: ignore
+                            next_request,
+                            privacy_request.get_cached_identity_data(),
+                            read_request,
+                        )
+                        rows.extend(
+                            self._apply_output_template(
+                                [param_value_map],
+                                read_request.output,
+                                processed_rows,
+                            )
+                        )
+
+            # This allows us to build an output object even if we didn't generate and execute
+            # any HTTP requests. This is useful if we just want to select specific input_data
+            # values to provide as row data to the mask_data function
+            if not read_request.path and read_request.output:
+                rows.extend(
+                    self._apply_output_template(
+                        query_config.generate_param_value_maps(
+                            input_data, read_request
+                        ),
+                        read_request.output,
                     )
-                    rows.extend(processed_rows)
+                )
+
         self.unset_connector_state()
         if awaiting_async_callback:
             # If a read request was marked to expect async results, original response data here is ignored.
             # We'll instead use the data received in the callback URL later.
             # Raising an AwaitingAsyncTaskCallback to put this task in an awaiting_processing state
             raise AwaitingAsyncTaskCallback()
+
         return rows
 
+    def _apply_output_template(
+        self,
+        param_value_maps: List[Dict[str, Any]],
+        output_template: Optional[str],
+        processed_rows: Optional[List[Row]] = None,
+    ) -> List[Row]:
+        """
+        Applies the output template to each row in processed_rows or generates
+        rows from param values and the output template if no rows are provided.
+        """
+        if not output_template:
+            return processed_rows or []
+
+        result = []
+        for processed_row in processed_rows or [None]:  # type: ignore
+            for param_value_map in param_value_maps:
+                if processed_row:
+                    param_value_map[ALL_OBJECT_FIELDS] = processed_row
+                row = assign_placeholders(output_template, param_value_map)
+                result.append(json.loads(row))  # type: ignore
```

Comment:
is this fully insured that it's valid, de-serializable JSON here?  do we perhaps need some validation on the `output` field for the read request? (i'll leave a comment up there too)